### PR TITLE
Topic/reqs 942

### PR DESCRIPTION
**Pull request by yukiiwashita**
* Added a demo test **EquityIndexOptionSmileAndPriceGreeksE2ETest** which includes smile construction, price/Greeks computation, and vol shift. 
* The corresponding rst is **worked-out-example-smile-equity-index-sparse-data.rst**.

### CODE REVIEW

File: docs/og-analytics/worked-out-examples/worked-out-example-smile-equity-index-sparse-data.rst [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+Given a set of information on the underlying,  
+
+.. code-block:: java 
+
+    double SHARES = 10.0;
+    int DAYS_TO_MAT = 23;
+    double FORWARD = 3360;
+    double RATE = 0.204 * 0.01;
+    double DIVIDEND = 0.333 * 0.01;
```
Comment (marc-opengamma) [inline docs/og-analytics/worked-out-examples/worked-out-example-smile-equity-index-sparse-data.rst]: Explain what RATE and DIVIDEND numbers mean. Why 0.204 * 0.01?

File: docs/og-analytics/worked-out-examples/worked-out-example-smile-equity-index-sparse-data.rst [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+.. code-block:: java 
+
+  double[] PUT_IV;
+  PUT_IV[i] = BlackFormulaRepository.impliedVolatility(PUT_PRICES[i] / DF, FORWARD, PUT_STRIKES[i], EXPIRY, false);
+
+Then the smile construction methodologies are applied to the implied volatilities. 
+
+In general market prices of call options are inconsistent with those of put options in the sense call and put with the same strike lead to different values of implied volatility, which is mainly a consequence of liquidity difference
+between the call and put options, and also between options and underlying.
```
Comment (marc-opengamma) [inline docs/og-analytics/worked-out-examples/worked-out-example-smile-equity-index-sparse-data.rst]: What about discrete dividend impact?

File: docs/og-analytics/worked-out-examples/worked-out-example-smile-equity-index-sparse-data.rst [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+------------------
+
+
+The global SABR method for the call options is called in the method **fitSabrSmileCall**. First we instantiate a **BitSet**
+
+.. code-block:: java 
+
+    BitSet fixed = new BitSet();
+
+which is used for fixing a specific SABR parameter. In order to fix the beta parameter, we set
```
Comment (marc-opengamma) [inline docs/og-analytics/worked-out-examples/worked-out-example-smile-equity-index-sparse-data.rst]: This is switch off in the code. The resulting calibration is probably unstable.

File: projects/OG-Analytics/src/test/java/com/opengamma/analytics/financial/model/volatility/smile/fitting/demo/EquityIndexOptionSmileAndPriceGreeksE2ETest.java [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+    //    printDetailsWithShift(smile, DISPLAY_STRIKES, true, VOL_SHOCKS); // vol shocks
+  }
+
+  /**
+   * Put options, global SABR.
+   */
+  @Test(description = "Demo", enabled = false)
+  public void fitSabrSmilePut() {
+    BitSet fixed = new BitSet();
+    // fixed.set(1); // turn on if beta value is fixed
```
Comment (marc-opengamma) [inline projects/OG-Analytics/src/test/java/com/opengamma/analytics/financial/model/volatility/smile/fitting/demo/EquityIndexOptionSmileAndPriceGreeksE2ETest.java]: Why switch off?

File: docs/og-analytics/worked-out-examples/worked-out-example-smile-equity-index-sparse-data.rst (hunk 1 above)
Comment (yukiiwashita) [inline docs/og-analytics/worked-out-examples/worked-out-example-smile-equity-index-sparse-data.rst]: Comments added.

File: docs/og-analytics/worked-out-examples/worked-out-example-smile-equity-index-sparse-data.rst (hunk 2 above)
Comment (yukiiwashita) [inline docs/og-analytics/worked-out-examples/worked-out-example-smile-equity-index-sparse-data.rst]: Comments added.

File: docs/og-analytics/worked-out-examples/worked-out-example-smile-equity-index-sparse-data.rst (hunk 3 above)
Comment (yukiiwashita) [inline docs/og-analytics/worked-out-examples/worked-out-example-smile-equity-index-sparse-data.rst]: Set beta fixed by default.

**Pull request merged**